File: legacy/legacy-apis/Psi.Models.ClientConfigurationModels/MobileConfigurations/Deals/CheckingRewards/CheckingRewards.cs
using System;

namespace Psi.Data.Models.ClientConfigurationModels.MobileConfigurations.Deals.CheckingRewards
{
    public class CheckingRewards : SettingsBaseHelper

    {
        private Authentication.Authentication _authentication;

        public CheckingRewards(ISettingsBase settingsBase) : base(settingsBase)
        {
        }

        [SettingKey("Mobile.Deals.CheckingRewards.MinimumVersion")]
        public string MinimumVersion
        {
            get { return GetValue(); }
            set { SetValue(value); }
        }

        [SettingKey("Mobile.Deals.CheckingRewards.MinimumAndroidVersion")]
        public string MinimumAndroidVersion
        {
            get { return GetValue(); }
            set { SetValue(value); }
        }

        [SettingKey("Mobile.Deals.CheckingRewards.MinimumIosVersion")]
        public string MinimumIosVersion
        {
            get { return GetValue(); }
            set { SetValue(value); }
        }

        [SettingKey("Mobile.Deals.CheckingRewards.Enabled")]
        public bool Enabled
        {
            get { return GetBoolValue(); }
            set { SetValue(value); }
        }

        [SettingKey("Mobile.Deals.CheckingRewards.Url")]
        public string Url
        {
            get { return GetValue(); }
            set { SetValue(value); }
        }

        public Authentication.Authentication Authentication
        {
            // The value is hardcoded to match the feature public id in the Meta.LayeredSecurityFeature table
            get { return _authentication ?? (_authentication = new Authentication.Authentication(new Guid("168A15A6-D8FD-4E73-AF40-A78C0C3DC3F8"))); }
            set { _authentication = value; }
        }

        [SettingKey("Mobile.Deals.CheckingRewards.ReslovingUrl")]
        public string ReslovingUrl
        {
            get { return GetValue();
            }
            set { SetValue(value);
            }
        }
    }
}

File: workspace/legacy-analyzer/output/infrastructure/models/CheckingRewards/CheckingRewards.ts
// Generated imports
import { Authentication } from '../MobileConfigurations/Authentication/Authentication';

export interface CheckingRewards {
    /** @settingKey Mobile.Deals.CheckingRewards.MinimumVersion */
    minimumVersion: string;
    /** @settingKey Mobile.Deals.CheckingRewards.MinimumAndroidVersion */
    minimumAndroidVersion: string;
    /** @settingKey Mobile.Deals.CheckingRewards.MinimumIosVersion */
    minimumIosVersion: string;
    /** @settingKey Mobile.Deals.CheckingRewards.Enabled */
    enabled: boolean;
    /** @settingKey Mobile.Deals.CheckingRewards.Url */
    url: string;
    authentication: Authentication;
    /** @settingKey Mobile.Deals.CheckingRewards.ReslovingUrl */
    reslovingUrl: string;
}
